test/DssVestTopUp.test.ts

```diff
@@ -86,199 +65,114 @@ describe("DssVestTopUp", function () {
-        );
-      });
 
-      it("should swap with slippage protection", async function () {
-        const refundTx = await topUp.refundUpkeep();
-
-        // get event data from mock
-        const refundRc = await refundTx.wait();
-        const callMockEvent = refundRc.events?.find(
-          (e) => e.address === swapRouterMock.address
-        );
-        const abiCoder = new ethers.utils.AbiCoder();
-        const [amountIn, amountOutMinimum] = abiCoder.decode(
-          ["uint256", "uint256"],
-          callMockEvent?.data || ""
-        );
-
-        const paymentLinkPrice = paymentUsdPrice / linkUsdPrice;
-        const linkAmtOut = amountIn * paymentLinkPrice;
-        const expectedLinkOutWithSlippage =
-          linkAmtOut - (linkAmtOut * slippageTolerancePercentage) / 100;
-
-        expect(amountOutMinimum).to.eq(expectedLinkOutWithSlippage);
-      });
+    it("should refund upkeep", async function () {
+      expect(await topUp.refundUpkeep())
+        .to.emit(topUp, "UpkeepRefunded")
+        .withArgs(topUpAmount);
     });
 
-    it("should fund with emergency topup", async function () {
-      await token.mint(topUp.address, minWithdrawAmt);
+    it("should update upkeep balance", async function () {
       await topUp.refundUpkeep();
 
-      const upkeepInfo = await keeperRegistryMock.getUpkeep(0);
-      expect(upkeepInfo.balance).to.eq(initialUpkeepBalance.add(100));
-    });
+      const upkeepInfo = await keeperRegistryMock.getUpkeep(fakeUpkeepId);
 
-    it("should fail if emergency topup below threshold", async function () {
-      await token.mint(topUp.address, 1);
-      await expect(topUp.refundUpkeep()).to.be.revertedWith(
-        "refund not needed"
-      );
+      expect(upkeepInfo.balance).to.eq(initialUpkeepBalance.add(topUpAmount));
     });
 
-    it("should not top up if not needed", async function () {
-      await expect(topUp.refundUpkeep()).to.be.revertedWith(
-        "refund not needed"
-      );
-    });
-  });
-
-  describe("Owner", async function () {
-    it("should update slippage tolerance percent", async function () {
-      const oldSlippagePercent = await topUp.uniswapSlippageTolerancePercent();
-      await topUp.setSlippageTolerancePercent(5);
-      const newSlippagePercent = await topUp.uniswapSlippageTolerancePercent();
-      expect(oldSlippagePercent).to.not.eq(newSlippagePercent);
-    });
+    it("should swap DAI for LINK", async function () {
+      // SwapRouterMock swaps 1-to-1
+      const expectedDaiAmountIn = topUpAmount;
+      const expectedLinkAmountOut = expectedDaiAmountIn;
 
-    it("should update pool fee", async function () {
-      const oldPoolFee = await topUp.uniswapPoolFee();
-      await topUp.setUniswapPoolFee(10000);
-      const newPoolFee = await topUp.uniswapPoolFee();
-      expect(oldPoolFee).to.not.eq(newPoolFee);
+      await expect(topUp.refundUpkeep())
+        .to.emit(topUp, "SwappedDaiForLink")
```

Comment:
Isnt it more reliable to check `balanceOf` of DaiToken ?

Reply:
It is but not sure if it's worth doing it for a mock contract.